### PR TITLE
refactor(rubygems): Unify fetching via v1 API

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

- Use single logic of fetching via `/v1/versions/foobar.json` with additional metadata in `/v1/gems/foobar.json`
- Disable caching for endpoints that meant to be non-standard.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
- Blocked by #23474

File: lib/modules/datasource/rubygems/index.ts [hunk 1]
```diff
@@ -91,102 +89,6 @@ export class RubyGemsDatasource extends Datasource {
...
-    }
-
-    if (gemMetadata.sourceUrl) {
-      result.sourceUrl = gemMetadata.sourceUrl;
-    }
-
-    return result;
-  }
-
-  @cache({
```
Comment (viceice) [inline lib/modules/datasource/rubygems/index.ts:183]: why cache removal?

**Review by viceice — COMMENTED** [review]
why do we loose the package caching? otherwise LGTM

File: lib/modules/datasource/rubygems/index.ts (hunk 1 above)
Comment (zharinov) [reply lib/modules/datasource/rubygems/index.ts:183]: Rubygems data is cached in separate place, and we don't want cache other registries, e.g. private ones

**Comment by zharinov** [comment]
I have no idea how to fix coverage regression, Codecov isn't too helpful

**Review by viceice — APPROVED** [review]
I think we need a global flag to also cache private packages on self-hosted renovate. currently mostly need npm, nuget, helm, docker and terraform. 🙃

**Comment by zharinov** [comment]
Agreed. I think our caching is quite complex, and we rarely cover it in tests.

**Pull request merged**